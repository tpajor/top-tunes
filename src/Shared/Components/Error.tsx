import * as React from 'react';

interface IProps {
  error: string;
}

const Error = ({error}: IProps) => {
  return (
    <h4 className='text text--error'>
      {`${error}`}<br />
      Please try again later
    </h4>
  );
};

export default Error;
