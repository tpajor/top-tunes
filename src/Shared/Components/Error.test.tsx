import * as React from 'react';
import { shallow } from 'enzyme';

import Error, { IProps } from './Error';

function setup() {
  const props: IProps = {
    error: 'error',
  };

  const wrapper = shallow(<Error {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('Error component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
