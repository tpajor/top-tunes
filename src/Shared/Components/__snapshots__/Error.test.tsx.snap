// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    error="error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "error",
        <br />,
        "Please try again later",
      ],
      "className": "text text--error",
    },
    "ref": null,
    "rendered": Array [
      "error",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "Please try again later",
    ],
    "type": "h4",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "error",
          <br />,
          "Please try again later",
        ],
        "className": "text text--error",
      },
      "ref": null,
      "rendered": Array [
        "error",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Please try again later",
      ],
      "type": "h4",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
